docs/reference-spec.md

```diff
@@ -0,0 +1,64 @@
+# Reference and Naming Specification
+
+This section of the Bindle specification defines how naming and referencing works.
+
+## Bindle Name (and Version)
+
+A bindle has a name. For example, if we were to write a `hello world` bindle, it may have the name `hello_world`.
+
+When a bindle is concretized into an invoice, it MUST have a name and a version. Version MUST be a valid [https://semver.org](SemVer 2 version).
+
+A bindle name can be one or more characters from the set of unicode alphabetic and numeric characters, as well as the characters `_` and `/`.
+
+TODO: Are there other characters we should allow? `-` causes problems with algorithms that autoconvert between `_` and `-`, because of ambiguity. `+` and `%` are legal in URLs, and % might be useful as an escape character, per URLs. A bindle name MUST NOT start with the string `bindle:` (the `:` is not a legal character in bindle names).
```

Comment:
Is the "A bindle name MUST NOT` section part of this TODO? It is unclear

Reply:
It was, yes -- as a clarification on the note. Since `:` is not a legal character in the name field, I didn't think this sentence needed to be included in the spec. I could bump it out of the TODO, though.

Reply:
Don't have a super strong opinion on this either way